general/src/protocol/contexts.rs

```diff
@@ -667,6 +667,32 @@ impl Contexts {
             self.insert(key.to_owned(), Annotated::new(ContextInner(context)));
         }
     }
+
+    /// Adds a context at the specified index,
+    /// Prefer [add] when adding contexts with valid default_key()
+    /// This method should be used for Other context or for contexts that are
+    /// not added at their default keys
+    pub fn add_at_index<S>(&mut self, key: S, context: Context)
```

Comment:
Usually, this method is called `insert` for dictionary-like objects in Rust. If you think this is too ambiguous, we could rename `add` into `push` to distinguish it better.

```suggestion
    pub fn insert<S>(&mut self, key: S, context: Context)
```